tests/queries/0_stateless/02784_parallel_replicas_automatic_decision.sh

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env bash
+
+CUR_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CUR_DIR"/../shell_config.sh
+
+function involved_parallel_replicas () {
+    # Not using current_database = '$CLICKHOUSE_DATABASE' as nested parallel queries aren't run with it
+    $CLICKHOUSE_CLIENT --query "
+        SELECT
+            initial_query_id,
+            (count() - 2) / 2 as number_of_parallel_replicas
```

Comment:
Needs to be changed. See https://github.com/ClickHouse/ClickHouse/pull/54331

Reply:
Based on #54331 this method of checking how many parallel replicas were used in the query will be flaky, as the work might be finished by a fast replica before some other one finishes its setup, which means that it won't appear in the query_log. Instead I'll work on parsing the text log and extract the value and validate both the value and if parallel replicas where used or not.